File: stack.py
class Node:
    def __init__(self, data):
       self.data = data
       self.next = None
 
class Stack:
    def __init__(self):
        self.head = None
 
    def push(self, data):
        if self.head is None:
            self.head = Node(data)
        else:
            neNode = Node(data)
            newNode.next = self.head
            self.head = newNode
    def empty():
        return self.head == None
    def pop(self):
        if self.empty():
            raise NameError("Stack empty")
        else:
            elmt = self.head.data
            self.head = self.head.next
            return elmt

File: frontend/src/app/services/Structures/Stack.ts
import { Node } from './Node';
export class Stack {
    private top: Node;
    constructor(){
        this.top = null;

    }
    public push(elm: any): void{
        if (top == null) {
            this.top = new Node(elm);
        }else{
            const aux: Node = new Node(elm);
            aux.next = this.top;
            this.top = aux;
        }
    }
    public pop(): any{
        if (this.empty()) {
            console.error('Stack is empty');
        } else {
            console.log(this.top);
            const aux: Node = this.top;
            this.top = this.top.next;
            return aux.value;
        }
    }
    public empty(): boolean{
        return this.top == null;
    }
}
